### ISSUE TITLE
Enhancements to Manage Citations Modal

### ISSUE THREAD

**Issue by kjacobsen16** [USGS-WiM/NSS#197]
Both of these will be easier once 102 is merged into dev
- [x] Add edit button next to citations and regression regions

- [x] Add button for adding new citation

- [x]  Disable the edit button next to regressions not in their regions

**Comment by harper-wavra** [comment]
For the first point we currently have a button to edit the citation I am just wondering where a good spot would be for the button to edit a regression region? 

Then for the second point I just though a button near the top of the modal for adding a new citation

![image](https://user-images.githubusercontent.com/59063181/87588109-b7485e00-c6a8-11ea-9911-146deb60d3e4.png)

**Comment by kjacobsen16** [comment]
Hm, I forgot about this. Maybe we move the edit/delete citation buttons to the left of the citation row? And put the edit regression region button next to the left of each regression region? Maybe play around with it a bit and see what makes sense .

We could always wait for the edit regression region button in here. It was more of a nice lead in so you could remove the citation from the regression region easily before deleting it, but not necessary.

I think the add citation button sounds good!

**Comment by harper-wavra** [comment]
Right now the only way to add a new citation is inside the creating/editing a regression region modal. If we add a button for just creating a new citation inside the manage citation modal we (I think) won't want to use the creating/editing a regression region modal because the user wont be adding a regression region only a citation. 

I am wondering what the best way is to add a new citation without that modal. Would I need to create a new addcitation component with a modal for just the citation inputs? Or is there a way to have two separate modals inside the managecitations component (one for managing the citations and one for adding a citation)?

**Comment by kjacobsen16** [comment]
So, I actually misread your message at first and thought you were saying you can only create a citation by using the regression region in the url. So I looked that up and that's right, even though that's not what you were asking..

I don't know why it was set up that way, but looks like I'll have to add a separate endpoint allowing for just adding a citation before you can do this issue.

**Comment by harper-wavra** [comment]
I am having issues setting the region correctly in the edit regression region modal. The way we had it working before it just set the region to the selectedRegion which worked because you could only a edit a regression region after you had selected/were viewing the region it belongs too. Now however when you edit a regression region from the manage citation modal you could be in any of the regions so the wrong region populates in the edit regression region modal(if that makes sense.) I was trying to figure out if the region for the regression is in the response from the services but it looks like it is not, so I am unsure how to populate the region dropdown  in the edit regression region modal with the correct region from the regression region and not just the selectedRegion. Does anyone have any ideas?

**Comment by kjacobsen16** [comment]
I'm forgetting why the region dropdown is in the edit regression region modal.. Does that move the regression region to a different region, or copy it there? I'm thinking we could remove it from the edit regression region modal, but leave it when you're adding a new regression region.

If we do need it for some reason, I think I should be able to add the region to the regression region query in the services.
